compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2351,6 +2351,15 @@ class Typer extends Namer
         report.featureWarning(nme.dynamics.toString, "extension of type scala.Dynamic", cls, isRequired, cdef.srcPos)
       }
 
+      // @experimental inference
```

Comment:
Rather than inference, I would ask the user to add the annotation himself, being explicit is good here.

Reply:
Then ill change the warning to an error